infrastructure/lib/atat-web-api-stack.ts

```diff
@@ -269,7 +246,45 @@ export class AtatWebApiStack extends cdk.Stack {
       }).fn
     );
 
+    // The API spec, which just so happens to be a valid CloudFormation snippet (with some actual CloudFormation
+    // in it) gets uploaded to S3. The Asset resource reuses the same bucket that the CDK does, so this does not
+    // require any additional buckets to be created.
+    const apiAsset = new s3asset.Asset(this, "ApiSpecAsset", {
+      path: utils.packageRoot() + "../../atat_provisioning_wizard_api.yaml",
+    });
+    // AT-6769
+    const apiAsset2 = new s3asset.Asset(this, "ApiSpecAsset2", {
+      path: utils.packageRoot() + "../../atat_management_and_reporting_api.yaml",
+    });
+
+    // And now we include that snippet as an actual part of the template using the AWS::Include Transform. Since
+    // snippet is valid CloudFormation with real Fn::Sub invocations, those will be interpreted. This results in
+    // all of the function ARNs being inserted into the x-aws-apigateway-integration resources when the template
+    // is evaluated.
+    const apiSpecAsTemplateInclude = cdk.Fn.transform("AWS::Include", { Location: apiAsset.s3ObjectUrl });
+
+    const apiSpecAsTemplateInclude2 = cdk.Fn.transform("AWS::Include", { Location: apiAsset2.s3ObjectUrl });
+
+    // And with the data now loaded from the template, we can use ApiDefinition.fromInline to parse it as real
+    // OpenAPI spec (because it was!) and now we've got all our special AWS values and variables interpolated.
+    // This will get used as the `Body:` parameter in the underlying CloudFormation resource.
+    const apiGateway = new SecureRestApi(this, "AtatSpecTest", {
+      restApiName: `${props.environmentId} API`,
+      apiDefinition: apigw.ApiDefinition.fromInline(apiSpecAsTemplateInclude),
+      deployOptions: {
+        tracingEnabled: true,
+      },
+    }).restApi;
+
+    const apiGateway2 = new SecureRestApi(this, "AtatSpecTest2", {
+      restApiName: `${props.environmentId}2 API`,
+      apiDefinition: apigw.ApiDefinition.fromInline(apiSpecAsTemplateInclude2),
+      deployOptions: {
+        tracingEnabled: true,
+      },
+    }).restApi;
     this.restApi = apiGateway;
+    this.restApi2 = apiGateway2;
     this.addEmailRoutes(props);
     this.addTaskOrderRoutes(props);
     this.ssmParams.push(
```

Comment:
This should work just fine but to be clear, if this is really the route we want to go (two `SpecRestApi` resources) then we need to talk about how it impacts:
 - the UI (and its config endpoint)
 - CORS
 - the Systems Manager parameters we create

I'd _really_ prefer if we found a way to create a single file for this unless we truly expect to develop it as 2+ APIs over time. 

In any case, we should go ahead and plan for a [zero one infinity rule](https://en.wikipedia.org/wiki/Zero_one_infinity_rule) sort of thing here and support N API GW resources (and therefore use arrays and for loops).

Reply:
Looks like we're collapsing the two API specs into a single API spec, so only a single `SpecRestApi` will be required.  Thank you for the thoughtful input.